test/smoke/testApps/SpringBootAttachInMain/src/main/java/com/microsoft/applicationinsights/smoketestapp/AttachInFirstLineOfMain.java

```diff
@@ -0,0 +1,73 @@
+package com.microsoft.applicationinsights.smoketestapp;
+
+import com.microsoft.applicationinsights.agent.Agent;
+import net.bytebuddy.agent.ByteBuddyAgent;
+import org.springframework.lang.Nullable;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.management.ManagementFactory;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+public class AttachInFirstLineOfMain {
+
+    public static void attach() throws IOException {
+        Path tempDir = Files.createTempDirectory("applicationinsights");
+        Path agentJarFile = createTempAgentJarFile(tempDir);
+        Path configFile = createTempConfigFile(tempDir);
+        Runtime.getRuntime().addShutdownHook(new ShutdownHook(tempDir, agentJarFile, configFile));
+        ByteBuddyAgent.attach(agentJarFile.toFile(), getPid());
```

Comment:
Where is the hookup for agentmain in Agent.java from this line?

Reply:
check out https://www.baeldung.com/java-instrumentation